Repository/ServiceEntityRepository.php

```diff
@@ -19,6 +19,9 @@
  *         parent::__construct($registry, YourEntity::class);
  *     }
  * }
+ *
+ * @template T
+ * @template-extends EntityRepository<T>
```

Comment:
you might want to add `@psalm-param class-string<T> $entityClass` on the constructor (check the exact name of that special type. I may have made a mistake in it as I haven't checked the doc again about it when writing this comment).

Reply:
Nice idea, I added it. I checked and I think that's the right name of the type.